packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -341,10 +346,11 @@ export class MonacoEditorProvider {
         toDispose.push(editor.onLanguageChanged(() => this.updateMonacoDiffEditorOptions(editor)));
         return editor;
     }
-    protected createMonacoDiffEditorOptions(original: MonacoEditorModel, modified: MonacoEditorModel): MonacoDiffEditor.IOptions {
+    protected createMonacoDiffEditorOptions(original: MonacoEditorModel, modified: MonacoEditorModel, factoryOptions?: EditorFactoryOptions): MonacoDiffEditor.IOptions {
         const options = this.createOptions(this.diffPreferencePrefixes, modified.uri, modified.languageId);
         options.originalEditable = !original.readOnly;
         options.readOnly = modified.readOnly;
+        options.alwaysRevealFirst = Boolean(factoryOptions?.selection);
```

Comment:
This line is the actual fix. It needs to be known when the `DiffNavigator` is created because it's supposed to be `private` and inaccessible thereafter.